advtraj/cli/uclales.py

```diff
@@ -99,14 +105,23 @@ def main(data_path, file_prefix, output_path):
     ds = load_data(files=files, fields_to_keep=["w"])
 
     ds_ = ds.isel(time=int(ds.time.count()) // 2).sel(z=slice(300, None))
```

Comment:
I am a bit confused here. From my discussion with @leifdenby, I got the impression that @ReadingClouds 's changes were to not just expand advtraj to include multiple trajectories, but also to have tracers that start at different initial times? 
So is this line a simple example for a single trajectory that needs to be changed to select multiple time steps or am I missing something obvious?

Reply:
The first thing to note is that most of @ReadingClouds 's changes haven't been pulled yet. Including this pull request.

The second is that my original code, [https://github.com/ReadingClouds/Trajectories](https://github.com/ReadingClouds/Trajectories), documented currently at  [http://www.met.reading.ac.uk/~xm904103/code/trajectories/index.html]( http://www.met.reading.ac.uk/~xm904103/code/trajectories/index.html.), did a lot more than advtraj. advtraj (very sensibly) pulls out the core trajectory calculation and does nothing else apart from the valuable step of storing to netCDF.

The original advtraj code does work for multiple trajectories, but used different functions to do so. 
The changes in this pull are quite superficial really; Leif had time as a data_var, but it makes more sense to have it as a coord. I also introduce 'trajectory_number' as a coord to identify multiple trajectories, and 'ref_time' as a coord to identify the trajectory origin. The former made it possible to use the same code for single and multiple trajectories, while the latter enables mutiple trajectory origins.

Note that there is a slight change in philosophy here. My original code included the selection of start points. In advtraj, Leif took the very sensible approach that the same code should work for any set of (valid) start points, so providing them should be a separate task. Some examples are provided of how to do this and we shall expand that once we have the current queue of changes pulled. The bottom line is you need an Xarray Dataset containing data variables 'x', 'y' and 'z', coord 'time' (a single time); advtraj takes this and your LES output to produce a set of trajectories with 'ref_time' equal to the time of your initial points. The example just chooses the time in the 'middle' of the input data - it would be very straightforward to loop over multiple times.

I shall certainly be re-factoring my 'trajectory family' (i.e. multiple start times) code to use advtraj as soon I can and would be happy to discuss with you how that could be useful to you.